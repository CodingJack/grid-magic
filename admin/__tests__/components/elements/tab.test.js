/**
 * External dependencies.
 */
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

/**
 * Internal dependencies.
 */
import Tab from '../../../src/js/components/buttons/tab';

const namespace = 'esg-v4';
Enzyme.configure( { adapter: new Adapter() } );

test( 'it renders correctly', () => {
  const tree = create(
    <Tab 
      namespace={ namespace } 
      onClick={ () => {} } 
      section="overview"
    />
  ).toJSON();
  
  if( ! tree ) {
    return;
  }
  expect( tree ).toMatchSnapshot();
  const { props } = tree;

  props.icon = 'check';
  props.onClick = () => {};
  props.namespace = 'name-space';
  props.section = "editor";
  props.selected = true;

  expect( tree ).toMatchSnapshot();
} );

describe('onClick callback works and it unmounts', () => {
  let clicked;
  const component = shallow(
    <Tab 
      namespace={ namespace } 
      icon="check" 
      section="globals"
      onClick={ () => clicked = true } 
    /> 
  );

  it('should match the snapshot', () => {
    expect( component.html() ).toMatchSnapshot();
  });

  test( 'Click event', () => {
    component.simulate( 'click', {} );
    expect( clicked ).toEqual( true );
  } );

  test( 'it unmounts', () => {
    component.unmount();
  } );
} );




