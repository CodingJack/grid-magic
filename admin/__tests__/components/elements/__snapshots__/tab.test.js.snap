// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="esg-v4-error-js"
>
  <p>
    Something went wrong.  Please 
    <a
      href="https://github.com/themepunch/essential-grid-v4/issues"
    >
      report
    </a>
     the error below.
    <i>
      A Button Component requires text or an icon
    </i>
  </p>
  <p>
    <i>
      A Button Component requires text or an icon
    </i>
  </p>
</div>
`;

exports[`it renders correctly 2`] = `
<div
  className="esg-v4-error-js"
  icon="check"
  namespace="name-space"
  onClick={[Function]}
  section="editor"
  selected={true}
>
  <p>
    Something went wrong.  Please 
    <a
      href="https://github.com/themepunch/essential-grid-v4/issues"
    >
      report
    </a>
     the error below.
    <i>
      A Button Component requires text or an icon
    </i>
  </p>
  <p>
    <i>
      A Button Component requires text or an icon
    </i>
  </p>
</div>
`;

exports[`onClick callback works and it unmounts should match the snapshot 1`] = `"<button class=\\"esg-v4-btn esg-v4-white esg-v4-btn-icon-only esg-v4-menu-tab\\" role=\\"tab\\" tabindex=\\"0\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z\\"></path></svg></button>"`;
